internal/controllers/gateway/gateway_controller.go

```diff
@@ -582,6 +619,10 @@ func (r *GatewayReconciler) updateAddressesAndListenersStatus(
 		setGatewayCondition(gateway, readyCondition)
 		return true, r.Status().Update(ctx, pruneGatewayStatusConds(gateway))
 	}
+	if !reflect.DeepEqual(gateway.Status.Listeners, listenerStatuses) {
```

Comment:
We could NOT directly use `reflect.DeepEqual`  here, because the `listenerStatuses` comes from `getListenerStatus` that traverses through a map, and the order of elements in map is random, but `reflect.DeepEqual` compares the element on each place.

I think we could use either way to resolve this:
- change `getListenerStatus` to keep statuses of listeners in a fixed order
- compare the two slices by treating them as unordered sets

Reply:
9d59d6f8e8370c31d7863340ccf7b1e7cf857d45